src/gepa/api.py

```diff
@@ -38,11 +40,14 @@ def optimize(
     # Merge-based configuration
     use_merge=False,
     max_merge_invocations=5,
-    # Budget
+    # Budget and Stop Condition
     max_metric_calls=None,
+    stop_callback: Callable[[], bool] | None = None,
+    enable_signal_handling: bool = True,
```

Comment:
Can we remove this flag. I don't want signal handling logic anywhere in api.py, or core/engine.py. Is there a good way to avoid it?